### DISCUSSION TITLE
Feat: add support for Deepseek model

### DISCUSSION THREAD

**Discussion by zeroaddresss** [olimorris/codecompanion.nvim#113] [Ideas]
Ollama offers the possibility to use Deepseek-coder-v2 model (one of the cheapest and best performing models for coding), but the 236B model is really heavy to be run locally. However, the deepseek API allows to use it, and it has great results as a code assistant.
It would be awesome if it could be integrated within the codecompanion plugin.

**Comment by olimorris** [comment]
I wasn't aware of this. Very interesting and thanks for sharing.

I'd absolutely be open to a PR on this. I've written a [guide](https://github.com/olimorris/codecompanion.nvim/blob/main/doc/ADAPTERS.md) on how to create a adapter if you'd like to make a PR for this.

**Comment by SDGLBL** [comment]
@zeroaddresss 

You don't need to implement a new adapter because Deepseek's API is fully compatible with OpenAI's API. You just need to add the configuration like this:

```lua
require("codecompanion").setup({
  adapters = {
    deepseek = require("codecompanion.adapters").extend("openai", {
      env = {
        api_key = "cmd:gpg --decrypt ~/.deepseek-api-key.gpg 2>/dev/null",
      },
      url = os.getenv("DEEPSEEK_API_BASE") .. "/chat/completions",
      schema = {
        model = {
          default = "deepseek-chat",
          choices = {
            "deepseek-coder",
            "deepseek-chat",
          },
        },
        max_token = {
          default = 8192,
        },
        temperature = {
          default = 1,
        },
      },
    }),
  },
})
```

**Reply by richtong** [reply]
Wow this is cool doing it now

**Reply by richtong** [reply]
fyi this works great for deepseek-chat v2.5 that merges chat and coder into one model. If you specify 8192 you get the beta model as well that supports it.

And note you can always run the older deepseek v2 in ollama as well.

https://ollama.com/library/deepseek-v2
